src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -166,18 +167,37 @@ async function addToCart({ params, request, locals }: RequestEvent) {
 			chosenVariations: z.record(z.string(), z.string()).optional()
 		})
 		.parse(json);
+	const variationPrice =
+		product.hasVariations && chosenVariations
+			? sumCurrency(
+					runtimeConfig.mainCurrency,
+					Object.entries(chosenVariations).map((variation) => ({
+						amount:
+							product.variations?.find(
+								(vari) => variation[0] === vari.name && variation[1] === vari.value
+							)?.price ?? 0,
+						currency: runtimeConfig.mainCurrency
+					}))
+			  )
+			: 0;
```

Comment:
This should be moved to the `addToCartInDb` function, so the logic is present for each caller of the function and does not need to be duplicated

Also, in that function, we should make sure that:

- Only one variation per category is chosen
- At least One variation for each caetgory is chosen

If a variation has a negative price offset, it will prevent a user from adding that variation many times to reduce the price

Reply:
Minimum variation price = 0

Reply:
> Minimum variation price = 0

I'm not sure what you mean

From https://github.com/B2Bitcoin/beBOP/issues/1472 we can see in the screenshot a price delta of -10 for the first variation.

If a user adds the same variation many times they could bring down the total price of the product to 0 no?

Reply:
We can no more have a price delta of -10

Reply:
price delta minimum = 0 when we create product

Reply:
By the way, here you set `customPrice = sum(variations)`, it should be `price = sum(originalPrice, variations)` no?

Even if we cannot have variations with negative price deltas (@Tirodem ?) we should still enforce integrity of variation choices.

And the rest of the comment (moving the logic to the `addToCartInDb`) still holds